src/NServiceBus.AcceptanceTests/Routing/When_doing_content_based_routing_with_commands.cs

```diff
@@ -0,0 +1,139 @@
+                        await session.Send(new MyCommand { Instance = Discriminator1 });
+                        await session.Send(new MyCommand { Instance = Discriminator2 });
+                        await session.Send(new MyCommand { Instance = Discriminator1 });
+                        await session.Send(new MyCommand { Instance = Discriminator2 });
+                    })
+                )
+                .WithEndpoint<Receiver>(b => b.CustomConfig(cfg => cfg.MakeInstanceUniquelyAddressable(Discriminator1)))
+                .WithEndpoint<Receiver>(b => b.CustomConfig(cfg => cfg.MakeInstanceUniquelyAddressable(Discriminator2)))
+                .Done(c => c.MessageDeliveredReceiver1 >= 2 && c.MessageDeliveredReceiver2 >=2)
+                .Run();
+
+            Assert.AreEqual(2, ctx.MessageDeliveredReceiver1);
+            Assert.AreEqual(2, ctx.MessageDeliveredReceiver2);
+        }
+
+        static string Discriminator1 = "553E9649";
+        static string Discriminator2 = "F9D0022C";
+
+        static string ReceiverEndpoint = Conventions.EndpointNamingConvention(typeof(Receiver));
+
+        public class Context : ScenarioContext
+        {
+            public int MessageDeliveredReceiver1;
+            public int MessageDeliveredReceiver2;
+        }
+
+        public class Sender : EndpointConfigurationBuilder
+        {
+            public Sender()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.GetSettings().GetOrCreate<UnicastRoutingTable>()
+                        .AddOrReplaceRoutes("CustomRoutingFeature", new List<RouteTableEntry>
+                        {
+                            new RouteTableEntry(typeof(MyCommand), UnicastRoute.CreateFromEndpointName(ReceiverEndpoint))
+                        });
+                    c.GetSettings().GetOrCreate<EndpointInstances>()
+                        .AddOrReplaceInstances("CustomRoutingFeature", new List<EndpointInstance>
+                        {
+                            new EndpointInstance(ReceiverEndpoint, Discriminator1),
+                            new EndpointInstance(ReceiverEndpoint, Discriminator2)
+                        });
+                    c.GetSettings().GetOrCreate<DistributionPolicy>()
+                        .SetDistributionStrategy(new ContentBasedRoutingStrategy(ReceiverEndpoint));
+                });
+            }
+
+            class ContentBasedRoutingStrategy : DistributionStrategy
+            {
+                public ContentBasedRoutingStrategy(string endpoint) : base(endpoint, DistributionStrategyScope.Send)
+                {
+                }
+
+                public override string SelectReceiver(string[] receiverAddresses)
+                {
+                    throw new NotImplementedException(); // should never be called
+                }
+
+                public override string SelectDestination(string[] receiverAddresses, IOutgoingContext outgoingContext)
```

Comment:
Discussed with @timbussmann and @andreasohlund, we came to the conclusion that a distribution strategy context is better than passing in `IOutgoingContext`.